### PR TITLE
Fix docs for MCMC

### PR DESCRIPTION
**Pull request by neerajprad**
- add docs for `MCMC.run`.
 - adds `MCMCKernel` class to the docs.
 - updates documentation regarding default `num_warmup`.

### CODE REVIEW

File: pyro/poutine/handlers.py [hunk 1]
```diff
@@ -104,7 +104,7 @@ def handler(fn=None, *args, **kwargs):
             raise ValueError(
                 "{} is not callable, did you mean to pass it as a keyword arg?".format(fn))
         msngr = msngr_cls(*args, **kwargs)
-        return msngr(fn) if fn is not None else msngr
+        return functools.update_wrapper(msngr(fn), fn) if fn is not None else msngr
```
Comment (neerajprad) [inline pyro/poutine/handlers.py]: Without this, methods that use poutines as decorators will have their docstring reset. cc @eb8680 - does this seem fine?

File: pyro/infer/mcmc/api.py [hunk 2]
```diff
@@ -350,6 +350,26 @@ def __init__(self, kernel, num_samples, warmup_steps=None, initial_params=None,
...
+
+            nuts_kernel = NUTS(model)
+            mcmc = MCMC(nuts_kernel, num_samples=500)
+            mcmc.run(data)
+            samples = mcmc.get_samples()
+
+        :param args: optional arguments taken by
+            :meth:`MCMCKernel.setup <pyro.infer.mcmc.mcmc_kernel.MCMCKernel.setup>`.
+        :param kwargs: optional keywords arguments taken by
+            :meth:`MCMCKernel.setup <pyro.infer.mcmc.mcmc_kernel.MCMCKernel.setup>`.
```
Comment (fritzo) [inline pyro/infer/mcmc/api.py]: Oh nice, I was unfamiliar with this syntax, but will start using it!

File: pyro/poutine/handlers.py (hunk 1 above)
Comment (eb8680) [reply pyro/poutine/handlers.py]: LGTM

Comment (neerajprad) [reply pyro/poutine/handlers.py]: Hmm..seems like this is breaking something - `make doctest` and `tests.poutine.test_poutines.test_decorator_interface_primitives` seem to fail. I'll investigate and update soon.

**Comment by fehiepsi** [comment]
Thanks, @neerajprad !

**Pull request merged**